src/utils/hooks.jsx

```diff
@@ -22,17 +24,38 @@ export const useRenderContactHelpText = (enterpriseConfig) => {
   return renderContactHelpText;
 };
 
+export const useAlgoliaSearchApiKey = (config) => {
+  // If the search API key is not provided in the config,
+  // fetch it from `ALGOLIA_SECURED_KEY_ENDPOINT`.
+
+  const [searchApiKey, setSearchApiKey] = useState(config.ALGOLIA_SEARCH_API_KEY);
+
+  useEffect(() => {
+    const fetchApiKey = async () => {
+      const key = await fetchAlgoliaSecuredApiKey();
+      setSearchApiKey(key);
+    };
+
+    if (!config.ALGOLIA_SEARCH_API_KEY) {
+      fetchApiKey();
+    }
+  }, [config.ALGOLIA_SEARCH_API_KEY]);
+
+  return searchApiKey;
+}
```

Comment:
In case ` ALGOLIA_SEARCH_API_KEY` is null, I believe the call to the backend is made multiple times (each time the hook is used in a component) and the result is always the same. Maybe make this more global  so that the call is made only once (one-off hook in App.jsx or a global var)? 

Furthermore, since the key doesn't expire as per the current [implementation](https://github.com/open-craft/edx-enterprise/pull/11/files#diff-ed89653d794c1834d4d3043cd4ef9bbd73abd8f5c631da219e40cbddaa37cc37R549), why not use persist it in local storage to avoid the call the to the backend altogether?

Reply:
Ah right, @CefBoud, I'll check how to make this call only once.

>Furthermore, since the key doesn't expire as per the current [implementation](https://github.com/open-craft/edx-enterprise/pull/11/files#diff-ed89653d794c1834d4d3043cd4ef9bbd73abd8f5c631da219e40cbddaa37cc37R549), why not persist it to local storage to avoid the call the to the backend altogether?

I thought about this, but how would we know if a user assigned to a new / additional enterprise customer? In this case, we'll have to invalidate the cache, since all Algolia filters are encoded in the secured key.